File: easy/armstrong.lua
function div(a, b) return math.floor(a/b) end

for line in io.lines(arg[1]) do
  local a = tonumber(line)
  local e, r = math.floor(math.log10(a)+1), a
  while a > 0 do
    r, a = r - math.pow(a % 10, e), div(a, 10)
  end
  print((r == 0) and "True" or "False")
end

File: easy/armstrong.clj
(defn armstrong [a]
  (def e (Math/ceil (Math/log10 a)))
  (def i a)
  (def r a)
  (while (> i 0)
    (when true
      (def r (- r (Math/pow (mod i 10) e)))
      (def i (quot i 10))))
  (if (== r 0)
    "True"
    "False"))

(with-open [rdr (clojure.java.io/reader (first *command-line-args*))]
  (doseq [line (line-seq rdr)]
    (println (armstrong (Integer/parseInt line)))))
